docs/ad/api.md

```diff
@@ -1108,60 +1443,153 @@ POST _opendistro/_alerting/monitors
     },
     {
-      "model_id" : "cneh7HEBHPICjJIdXdrR_model_threshold",
-      "node_id" : "Og23iUroTdKrkwS-y89zLw"
+      "model_id": "cneh7HEBHPICjJIdXdrR_model_threshold",
+      "node_id": "Og23iUroTdKrkwS-y89zLw"
     },
     {
-      "model_id" : "cneh7HEBHPICjJIdXdrR_model_rcf_0",
-      "model_size_in_bytes" : 4456448,
-      "node_id" : "Og23iUroTdKrkwS-y89zLw"
+      "model_id": "cneh7HEBHPICjJIdXdrR_model_rcf_0",
+      "model_size_in_bytes": 4456448,
+      "node_id": "Og23iUroTdKrkwS-y89zLw"
     }
   ],
-  "shingle_size" : 8,
-  "coordinating_node" : "Og23iUroTdKrkwS-y89zLw",
-  "total_size_in_bytes" : 13369344
+  "shingle_size": 8,
+  "coordinating_node": "Og23iUroTdKrkwS-y89zLw",
+  "total_size_in_bytes": 13369344,
+  "init_progress": {
+    "percentage": "70%",
+    "estimated_minutes_left": 77,
+    "needed_shingles": 77
+  }
 }
 
-GET _opendistro/_anomaly_detection/detectors/m4ccEnIBTXsGi3mvMt9p/_profile/total_size_in_bytes
+GET _opendistro/_anomaly_detection/detectors/<detectorId>/_profile/total_size_in_bytes
 
 {
   "total_size_in_bytes" : 13369344
 }
 ```
 
+If you have configured the category field, you can see the number of unique values in the field and also all the active entities with models running in memory.
+You can use this data to estimate the memory required for anomaly detection to help decide the size of your cluster.
+For example, if a detector has one million entities and only 10 of them are active in memory, then you need to scale up your cluster.
+
+#### Request
+
+```json
+GET /_opendistro/_anomaly_detection/detectors/<detectorId>/_profile?_all=true&pretty
+
+{
+  "state": "RUNNING",
+  "models": [
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997684",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997685",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997686",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997680",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997681",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997682",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    },
+    {
+      "model_id": "T4c3dXUBj-2IZN7itix__entity_i-00f28ec1eb8997683",
+      "model_size_in_bytes": 712480,
+      "node_id": "g6pmr547QR-CfpEvO67M4g"
+    }
+  ],
+  "total_size_in_bytes": 4987360,
+  "init_progress": {
+    "percentage": "100%"
+  },
+  "total_entities": 7,
+  "active_entities": 7
+}
+```
+
```

Comment:
Can we add some transition sentences before the following text?  This is my original text in the quip doc:"Profile API also provides information about each entity, such as the entity’s last_sample_timestamp and last_active_timestamp. An entity with no anomaly result recorded can be due to either no data samples or the entity’s model being evicted out of the model cache. last_sample_timestamp tells the last document containing the entity, while last_active_timestamp tells the timestamp when the entity’s model is last seen in the model cache."